RTech.CoffeeMachine.Api.Tests/BrewCoffee/Controllers/CoffeeMachineControllerTests.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System.Net;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.Caching.Memory;
+using Moq;
+using RTech.CoffeeMachine.Api.BrewCoffee.Controllers;
+using RTech.CoffeeMachine.Api.BrewCoffee.Models;
+using RTech.CoffeeMachine.Api.BrewCoffee.Util;
+using Shouldly;
+
+namespace RTech.CoffeeMachine.Api.Tests.BrewCoffee.Controllers;
+
+public class CoffeeMachineControllerTests : IClassFixture<WebApplicationFactory<Program>>
+{
+    private readonly WebApplicationFactory<Program> _factory;
+
+    public CoffeeMachineControllerTests(WebApplicationFactory<Program> factory)
+    {
+        _factory = factory;
+    }
+
+    [Fact]
+    public async Task GetBrewStatus_WhenUnrecognizedBrewStatus_ShouldReturn500()
+    {
+        var brewStatusProvider = new Mock<IBrewStatusProvider>();
+        brewStatusProvider.Setup(p => p.GetBrewStatus())
+            .Returns(BrewStatus.None);
+
+        var response = await GetEndpointResponse(services =>
```

Comment:
`GetEndpointResponse` is a test helper method that creates a mocked instance of `WebApplicationFactory` & then calls the new `GET` endpoint (see bottom of this file)